vcf-dumper/vcf-dumper-lib/src/main/java/uk/ac/ebi/eva/vcfdump/VariantExporter.java

```diff
@@ -66,39 +70,53 @@ public VariantExporter() {
-                    variantsToExport.add(variantContext);
-                } catch (Exception e) {
-                    logger.warn("Variant {}:{}:{}>{} dump failed: {}", variant.getChromosome(), variant.getStart(),
-                                variant.getReference(),
-                                variant.getAlternate(), e.getMessage());
-                    failedVariants++;
+        try {
+            // TODO
+            List<VariantRepositoryFilter> filters = new FilterBuilder()
+                    .getVariantEntityRepositoryFilters(queryParams.getMaf(), queryParams.getPolyphenScore(),
+                                                       queryParams.getSiftScore(), queryParams.getStudies(), queryParams.getConsequenceType());
+
+            List<VariantWithSamplesAndAnnotation> variants = variantService.findByRegionsAndComplexFilters(
+                    Collections.singletonList(region), filters, null , null, null);
+
+            for (VariantWithSamplesAndAnnotation variant : variants) {
+                if (region.contains(variant.getChromosome(), variant.getStart())) {
+                    try {
+                        VariantContext variantContext = variantToVariantContextConverter.transform(variant);
+                        variantsToExport.add(variantContext);
+                    } catch (Exception e) {
+                        logger.warn("Variant {}:{}:{}>{} dump failed: {}", variant.getChromosome(), variant.getStart(),
+                                    variant.getReference(),
+                                    variant.getAlternate(), e.getMessage());
+                        failedVariants++;
+                    }
                 }
             }
+        } catch (AnnotationMetadataNotFoundException e) {
+            e.printStackTrace();
         }
+
+
         return variantsToExport;
     }
 
-    public List<VariantSource> getSources(VariantSourceDBAdaptor sourceDBAdaptor,
-                                          List<String> studyIds, List<String> fileIds)
+    public List<VariantSource> getSources(VariantSourceService variantSourceService, List<String> studyIds, List<String> fileIds)
             throws IllegalArgumentException {
         
         // get sources
-        QueryOptions queryOptions = new QueryOptions();
-        if (fileIds != null && !fileIds.isEmpty()) {
-            queryOptions.put(VariantDBAdaptor.FILE_ID, fileIds);
-        }
-        List<VariantSource> sourcesList = sourceDBAdaptor.getAllSourcesByStudyIds(studyIds, queryOptions).getResult();
+//        QueryOptions queryOptions = new QueryOptions();
+//        if (fileIds != null && !fileIds.isEmpty()) {
+//            queryOptions.put(VariantDBAdaptor.FILE_ID, fileIds);
+//        }
+        Pageable pageable = new PageRequest(0, 1);
+
+        List<VariantSource> sourcesList = variantSourceService.findByStudyIdIn(studyIds,pageable );
```

Comment:
Why are we using a Pageable here? I think that we just needed it for getting the min or max start for the variants in a chromosome. Here we are getting the VariantSource object for every single study that is beinf exported. Adding the Pageabe, are we getting just one?

Reply:
I'm not sure how to get all available results, because Pageable is a parameter of `findByStudyIdIn`. Would passing null work in this case?